lib/util/package-rules.ts

```diff
@@ -194,7 +194,7 @@ function matchesRule(
   }
   if (matchSourceUrlPrefixes.length) {
     const isMatch = matchSourceUrlPrefixes.some((prefix) =>
-      sourceUrl?.startsWith(prefix)
+      sourceUrl?.toLowerCase().startsWith(prefix.toLowerCase())
```

Comment:
store `sourceUrl?.toLowerCase()` to a local const, so we don't need to do the conversion on every loop

Reply:
Updating, just a sec.

Reply:
Please add a test too

Reply:
Also please make the following changes:

- Edit the PR title to something like `fix: matchSourceUrlPrefixes compares case insensitive`
- Put `Fixes #10962` as text in the top comment, by doing that, GitHub will close the mentioned issue automatically once your PR gets merged. 😉 Read the [GitHub docs, linking a pull request to an issue using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) to learn more.